osutil/disks/disks_linux.go

```diff
@@ -408,7 +414,7 @@ func diskFromMountPointImpl(mountpoint string, opts *Options) (*disk, error) {
 
 func (d *disk) populatePartitions() error {
```

Comment:
(not releated to this PR but just an observation) - we cache the partitions over the entire lifetime of "Disk" which means if partitions get added or removed this will use outdated information. I wonder if we should document this (or lift the restriction and simply always repopulate). anyway, not an immediate concern just noticed :)

Reply:
Yes you're right we should document somewhere that the partitions calculated for a disk are only calculated once, they are not re-calculated. I have a slight preference to not lift the restriction given than the usage of Disk objects thus far has not been long enough to justify repopulating the partitions on every method call, and also that it would require more testing :smile:

Reply:
Added a TODO in https://github.com/snapcore/snapd/pull/10872